### ISSUE TITLE
Huffman table contains 256 entries (for no good reason)

### ISSUE THREAD

**Issue by SimonSegerblomRex** [thorfdbg/libjpeg#100]
**Problem**
Following this advice in the [README](https://github.com/thorfdbg/libjpeg/blob/master/README#L108):
> The following options exist for lossless coding integer:
> 
> predictive Rec. ITU-T T.81 | ISO/IEC 10918-1 coding. Note, however,
that not many implementations are capable of decoding such stream,
thus this is probably not a good option for all-day purposes.
> 
> $ jpeg -p -c infile.ppm out.jpg
> 
> While the result is a valid Rec. ITU-T T.81 | ISO/IEC 10918-1 stream,
most other implementations will hick up and break, thus it is not
advisable to use it.

libjpeg produces JPEG files that [libjpeg-turbo](https://github.com/libjpeg-turbo/libjpeg-turbo) refuses to decode with the error message
```
Bogus Huffman table definition
```
due to [this check](https://github.com/libjpeg-turbo/libjpeg-turbo/blob/main/jdhuff.c#L257) (discussed [here](https://github.com/libjpeg-turbo/libjpeg-turbo/issues/586)).

**Steps to reproduce**
Encode [this image](https://drive.google.com/file/d/10lETW_bcLgbUdjccmeAli7XAVPh5lzo1/view?usp=sharing) (or any(?) other PGM file) using libjpeg:
 ```sh
 ./jpeg -p -c 14bit.pgm out.jpg
 ```
Observe that the Huffman table has 256 entries.

**Workaround**
Encode the image using option `-h`:
```sh 
 ./jpeg -p -c -h 14bit.pgm out.jpg
```
Observe that the Huffman table has 10 entries.

**Reasoning**
One might argue that libjpeg-turbo is at fault not decoding this image since it's technically a valid JPEG file according to the specification, but the 256 entries table doesn't make sense either and seems to be bug in libjpeg. Also following up on [your comment here](https://github.com/libjpeg-turbo/libjpeg-turbo/issues/586#issuecomment-1079646085) @thorfdbg:

> If there are, for some image or some configuration, tables generated that populate more than the first 16 entries, please let me know because I would not know how this has happened (code for that in coding/huffmantemplate.cpp, lines 142 and following, val_dc_luminance/chrominance lists the symbols).

**Comment by SimonSegerblomRex** [comment]
I guess this might be a duplicate of https://github.com/thorfdbg/libjpeg/issues/88#issuecomment-1791584122... The documentation should be updated though.

**Comment by thorfdbg** [comment]
Thank you for reporting. Unlike the 12bit lossy case, I really do not recall in which mode the encoder could possibly need the additional symbols, so I removed them.

Fixed in 1.69, thank you.

**Comment by SimonSegerblomRex** [comment]
Thank you for the quick fix!
